core/src/ldml/ldml_transforms.cpp

```diff
@@ -462,9 +462,13 @@ transform_entry::init() {
     // TODO-LDML: if we have mapFrom, may need to do other processing.
     const std::u16string patstr = km::kbp::kmx::u32string_to_u16string(fFrom);
     UErrorCode status           = U_ZERO_ERROR;
-    /* const */ icu::UnicodeString patustr = icu::UnicodeString(patstr.data(), (int32_t)patstr.length());
+    /* const */ icu::UnicodeString patustr_raw = icu::UnicodeString(patstr.data(), (int32_t)patstr.length());
     // add '$' to match to end
-    patustr.append(u'$');
+    patustr_raw.append(u'$');
+    icu::UnicodeString patustr;
+    const icu::Normalizer2 *nfd = icu::Normalizer2::getNFDInstance(status);
+    // NFD normalize on pattern creation
+    nfd->normalize(patustr_raw, patustr, status);
     fFromPattern.reset(icu::RegexPattern::compile(patustr, 0, status));
```

Comment:
Do we need to assert status after each call?
* Yes, because we want to know exactly which call unravelled.
* Yes, because the input state to the following functions will be invalid -- or does icu check value of status variable?

Same thing applies for release builds we want to do _something_ if `!U_SUCCESS(status)`?

Reply:
Actually, when calling multiple ICU functions, we can actually not assert between calls, because each icu call is guaranteed to do nothing `if U_FAILURE(status)`.   All of these I would expect to fail under catastrophic situations.

Reply:
yes, it needs to propagate an error.

Reply:
OK, but even so it's probably beneficial to know which call failed?

Reply:
I think logging the failure code would be helpful also. And then should probably not expose the status to any of the consumers of the functions

Reply:
discussed DebugLog probably with a convenience macro to call `u_errorName(status)`